internal/sql/catalog/table.go

```diff
@@ -162,7 +163,7 @@ func (c *Catalog) createTable(stmt *ast.CreateTableStmt) error {
 			}
 			if col.Vals != nil {
 				typeName := ast.TypeName{
-					Name: col.Colname,
+					Name: fmt.Sprintf("%s_%s", stmt.Name.Name, col.Colname),
```

Comment:
One minor annoyance with this is that it doesn't "singularize" the table's name.

It might also be frustrating to always opt-in to the longer enum names, when sometimes the terser names might suffice. But that seems trickier to fix